src/utils.jl

```diff
@@ -55,45 +55,56 @@ function Base.last(sc::OperationalScenario)
     return ScenarioPeriod(
         sc.scen,
         sc.probability,
-        sc.multiple,
+        sc.mult_sc,
         last(sc.operational),
     )
 end
 
 function Base.last(rp::RepresentativePeriod)
     per = last(rp.operational)
-    mult = stripunit(rp.duration * rp.per_share / duration(rp.operational))
-    return ReprPeriod(rp.rper, per, mult)
+    return ReprPeriod(rp.rper, per, rp.mult_rp * multiple(per))
 end
 
 function Base.last(srp::StratReprPeriod)
     per = last(srp.operational)
-    mult = stripunit(srp.duration_sp * srp.op_per_strat / srp.duration_rp)
-    return OperationalPeriod(srp.sp, per, mult * multiple(per))
+    return OperationalPeriod(
+        srp.sp,
+        per,
+        srp.mult_sp * srp.mult_rp * multiple(per),
+    )
 end
 
 function Base.last(sp::StrategicPeriod)
     per = last(sp.operational)
-    mult = stripunit(duration(sp) * sp.op_per_strat / duration(sp.operational))
-    return OperationalPeriod(sp.sp, per, mult * multiple(per))
+    return OperationalPeriod(sp.sp, per, sp.mult_sp * multiple(per))
 end
 
 function Base.last(sos::StratOperationalScenario)
     per = last(sos.operational)
-    mult =
-        stripunit(sos.duration * sos.op_per_strat / duration(sos.operational))
-    return OperationalPeriod(sos.sp, per, mult * multiple(per))
+    return OperationalPeriod(sos.sp, per, sos.mult_sp * multiple(per))
+end
+
+function Base.last(sro::StratReprOpscenPeriod)
+    per = last(sro.operational)
+    rper = ReprPeriod(sro.rp, per, sro.mult_rp * multiple(per))
+    return OperationalPeriod(
+        sro.sp,
+        rper,
+        sro.mult_sp * sro.mult_rp * multiple(per),
+    )
 end
 
+#=
```

Comment:
I have tried to do a round of cleanup in the implementation of `last()`. As you say, this functionality is most relevant for iterable structures that are ordered, i.e `last(repr_periods(_))` and `last(strategic_periods(_)` while `last(opscenarios(_))` is not relevant. This is know supported in the code.

Similarly, `SimpleTimes`, `CalendarTimes`and `AbstractOperationalScenario` are naturally ordered and have `last()` implemented. The discussion point is whether an `AbstractRepresentativePeriod` should have `last()` implemented. Since a representative period can have multiple scenarios it will in general not be an ordered sequence of time periods. The current implementation is to have an error, and require code to go through `opscenarios(rp)` for each representative period and ensure cyclic conditions there. Another option is to allow `last()` for representative periods without operational scenarios and have an error otherwise.